main/src/ca/uwaterloo/flix/tools/pkg/ManifestParser.scala

```diff
@@ -16,22 +16,240 @@
+  def parse(p: Path)(implicit out: PrintStream): Result[Manifest, ManifestError] = {
+    val parser = try {
+      val parser = Toml.parse(p)
+      parser
+    } catch {
+      case _: IOException => return Err(ManifestError.IOError(p))
+    }
+
+    val errors = parser.errors
+    if(errors.size() > 0) {
+      var error_string = ""
+      errors.forEach(error => error_string = error_string + error.toString + ", ")
+      return Err(ManifestError.ManifestParseError(p, error_string))
+    }
+
+    val name = try {
+      val name = parser.getString("package.name")
+      if(name == null) {
+        return Err(ManifestError.RequiredPropertyIsNull(p, "'package.name' is null"))
+      }
+      name
+    } catch {
+      case _: IllegalArgumentException => return Err(ManifestError.MissingRequiredProperty(p, "'package.name' is missing"))
+      case _: TomlInvalidTypeException => return Err(ManifestError.RequiredPropertyHasWrongType(p, s"'package.name' should have type String"))
+    }
+
+    val description = try {
+      val description = parser.getString("package.description")
+      if (description == null) {
+        return Err(ManifestError.RequiredPropertyIsNull(p, "'package.description' is null"))
+      }
+      description
+    } catch {
+      case _: IllegalArgumentException => return Err(ManifestError.MissingRequiredProperty(p, "'package.description' is missing"))
+      case _: TomlInvalidTypeException => return Err(ManifestError.RequiredPropertyHasWrongType(p, s"'package.description' should have type String"))
+    }
+
+    val version = try {
+      val version = parser.getString("package.version")
+      if (version == null) {
+        return Err(ManifestError.RequiredPropertyIsNull(p, "'package.version' is null"))
+      }
+      version
+    } catch {
+      case _: IllegalArgumentException => return Err(ManifestError.MissingRequiredProperty(p, "'package.version' is missing"))
+      case _: TomlInvalidTypeException => return Err(ManifestError.RequiredPropertyHasWrongType(p, s"'package.version' should have type String"))
+    }
+    val semanticVersion = toSemVer(version, p) match {
+      case Ok(v) => v
+      case Err(e) => return Err(e)
+    }
+
+    val flix = try {
+      val flix = parser.getString("package.flix")
+      if (flix == null) {
+        return Err(ManifestError.RequiredPropertyIsNull(p, "'package.flix' is null"))
+      }
+      flix
+    } catch {
+      case _: IllegalArgumentException => return Err(ManifestError.MissingRequiredProperty(p, "'package.flix' is missing"))
+      case _: TomlInvalidTypeException => return Err(ManifestError.RequiredPropertyHasWrongType(p, s"'package.flix' should have type String"))
+    }
+    val semanticFlixVersion = toSemVer(flix, p) match {
+      case Ok(v) => v
+      case Err(e) => return Err(e)
+    }
+
+    val license_opt: Option[String] = try {
+      val license = parser.getString("package.license")
+      if (license == null) {
```

Comment:
In Scala we can just write `Option(license)` which is a short-hand for what you have with the `if`.